### PR TITLE
configs: add switch for tracking requests

### PR DESCRIPTION
**Pull request by sticnarf**
### What is changed and how it works?

Issue Number: ref #12362

What's Changed:

```commit-message
A dynamic config server.track-request is added to control whether
the metrics of each request should be tracked and returned to the
client.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM
- Breaking backward compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

File: src/server/service/kv.rs [hunk 1]
```diff
@@ -1567,12 +1581,17 @@ fn future_scan<E: Engine, L: LockManager, F: KvFormat>(
...
     storage: &Storage<E, L, F>,
     mut req: BatchGetRequest,
+    configs: &AtomicConfigs,
 ) -> impl Future<Output = ServerResult<BatchGetResponse>> {
-    let tracker = GLOBAL_TRACKERS.insert(Tracker::new(RequestInfo::new(
-        req.get_context(),
-        RequestType::KvBatchGet,
-        req.get_version(),
-    )));
+    let tracker = if configs.get_track_request() {
```
Comment (TonsnakeLin) [inline src/server/service/kv.rs:1586]: Could you wrapper the code block `let tracker = if {} else {}` to a function? I found several duplicated code.

Comment (sticnarf) [reply src/server/service/kv.rs:1586]: The types are different, so it's sadly impossible to extract the whole pattern as a function. And even the function names are not totally consistent, a macro cannot solve the problems either.

Comment (sticnarf) [reply src/server/service/kv.rs:1586]: Ah, if you only want to eliminate the if/else but still extract the fields manually, it's possible.

**Pull request closed without merging**